--- arj-3.10.22/fardata.c.org	2009-08-20 10:28:36.000000000 +0000
+++ arj-3.10.22/fardata.c	2009-08-20 10:28:53.000000000 +0000
@@ -190,7 +190,7 @@
 
 /* Length-limited strlen() */
 
-static int strnlen(const char FAR *s, int count)
+static int strnlen_arj(const char FAR *s, int count)
 {
  const char FAR *sc;
 
@@ -569,7 +569,7 @@
     if(!s)
      s="(null)";
 #endif
-    len=strnlen(s, precision);
+    len=strnlen_arj(s, precision);
     if(!(flags&LEFT))
     {
      while(len<field_width--)
